### ISSUE TITLE
Sourcemod 1.10 is stable. Need to update the Eventitem Spawner (if you got time)

### ISSUE THREAD

**Issue by Sacricx** [Totenfluch/EventItemsSpawner#1]
Since SM 1.10 is stable and in use on my servers, this plugin sadly won't work anymore. It would be pretty kind, if you gonna update this plugin since SM 1.10 is stable because you did a really really good job here and I'd love to continue using this awesome plugin on my servers. 
Currently works (but can't compile) with SM 1.9

This may help for a possible update/fix:
When I try to compile the plugin with Sourcemod 1.9 or 1.10 then I get the following error:
EventItemsSpawner.sp( 8 ): fatal error 183: cannot read from file: "autoexecconfig".

best regards

Sacricx

**Comment by Totenfluch** [comment]
[autoexecconfig.txt](https://github.com/Totenfluch/EventItemsSpawner/files/3944687/autoexecconfig.txt)

**Comment by Sacricx** [comment]
Thank you very much for providing the autoexecconfig.txt !
Since I can only guess where to put this in, I tried all diffrent folders in addons/sourcemod/scripting/ 
also in addons/sourcemod/. But no matter where I put the autoexecconfig.txt the error shows up on compile. I would appreciate very much, if you could tell me/people who want to know, where to put this cfg. That would be cool :) !

best regards

Sacricx

**Comment by Totenfluch** [comment]
rename it to autoexecconfig.inc

and it's /scripting/includes

**Comment by Sacricx** [comment]
Although this issue is closed already I want to share my solution of how to compile with Sourcemod 1.10 since adding the autoexecconfig.inc was just the first step which was neccessary successfully compile the plugin.

1. You need to add smlib to your includes folder but download the traditional syntax branch since the master branch uses deprecated syntax and won't let you compile against SM 1.10.

DOWNLOAD HERE: https://github.com/bcserv/smlib/tree/transitional_syntax


2. Make sure to add: emitsoundany.inc, map_workshop_functions.inc, multicolors.inc, colors.inc, morecolors.inc store.inc to your include folder.

DOWNLOAD HERE: 

- https://forums.alliedmods.net/showthread.php?t=237045 (emitsoundany.inc)
- https://forums.alliedmods.net/showthread.php?t=249866 (map_workshop_functions.inc)
- https://github.com/Bara/Multi-Colors (multicolors.inc | colors.inc and morecolors.inc come with this)
- https://github.com/dvarnai/store-plugin (store.inc | If you already have the store added you can ignore this)


3. Edit store.inc

REMOVE THE COMPLETE LINES:

- String:szName[ITEM_NAME_LENGTH], (Line 24)
- iPrice, (Line 28)
- iId, (Line 55)


The plugin should compile successfully with 6 warnings which you can ignore for now. 
I hope this detailed solution and report helps someone who wants to compile against SM 1.10.


4. To Totenfluch:

This warnings are related but can be ignored until SM 1.11 I guess:

- addons\sourcemod\scripting\EventItemsSpawner.sp(32) : warning 241: Array-based enum structs will be removed in 1.11. 
See https://wiki.alliedmods.net/SourcePawn_Transitional_Syntax#Enum_Structs

- addons\sourcemod\scripting\EventItemsSpawner.sp(278) : warning 213: tag mismatch

- addons\sourcemod\scripting\EventItemsSpawner.sp(6) : warning 203: symbol is never used: "g_cvarChatTag"

**Comment by Totenfluch** [comment]
You are basically describing how to compile the plugin. This does not seem to be an issue with 1.10 in particular just you having to go through the though process of compiling this. I could have done a better job documenting this bu that's why there's a .smx in the directory.

about the 1.11 yes I agree this is problematic but since the whole plugin relies on enum-structs this will be a though one to migrate

**Comment by Totenfluch** [comment]
https://github.com/Totenfluch/EventItemsSpawner/commit/b8eccbfa0df0a2d0005e0c411620124ef97b0342

**Comment by Sacricx** [comment]
You are right its no issue with SM 1.10. Without the include files you can't compile it against SM 1.9 either. That was not necessary for me in the past tho since the provided .smx in the directory worked well for me with SM 1.9. Because of that I did never try to recompile with SM 1.9 and thought it's been an issue with SM 1.10.

Unfortunatley the provided .smx doesn't work with SM 1.10 even when the cfg's are set properly 
(no errorlogs tho). That's why I wanted to recompile the .sp file. Once I looked at the errors, I got rid of them step by step. And yes I could have made this alot easier by requesting you to add all the necessary include files but to edit the store.inc before.

The whole reason for my issue was simply that some plugins I'm using require SM 1.10 meanwhile which forced me to update to SM 1.10. But your plugin is so nice that I didn't want to miss it on my servers <3

Yeah and since I had no guarantee, that you come back to this issue at anytime in the future I wanted to make sure to let people know about my "solution" to this and described for them how to be able to compile against SM 1.10.

I do appreciate your reply very much and want to say that its still nowadays an awesome plugin for my Community. It was a really great idea from you to invent it and we don't want to miss it on our servers. 

So long story short, thank you very much for adding the required include files to the master branch.


I wish you a good time mate :) !

Sacricx
